### PR TITLE
Refactor scm status reporter

### PR DESCRIPTION
**Pull request by danidoni**
This is a follow up on #12625 

This PR tries to reduce the responsibilities of the `SCMStatusReporter` by splitting it in two classes: `GithubStatusReporter` and `GitlabStatusReporter` that each one knows how to report against it's own SCM

### CODE REVIEW

**Comment by danidoni** [comment]
[Refactor the ScmStatusReporter#call method](https://trello.com/c/m2pZJsY0/148-refactor-the-scmstatusreportercall-method)

File: src/api/app/services/github_status_reporter.rb
Comment (dmarcoux) [inline src/api/app/services/github_status_reporter.rb:30]: This code is now specific to GitHub, so this comment (including the next line) should be removed. It doesn't provide value.

File: src/api/app/services/scm_status_reporter.rb
Comment (dmarcoux) [inline src/api/app/services/scm_status_reporter.rb:12]: We should instead raise `AbstractMethodCalled`:

```suggestion
    raise AbstractMethodCalled
```

File: src/api/app/services/scm_status_reporter.rb
Comment (dmarcoux) [inline src/api/app/services/scm_status_reporter.rb:11]:
```suggestion
    # Should use either GithubStatusReporter or GitlabStatusReporter
```

File: src/api/app/services/scm_status_reporter.rb
Comment (dmarcoux) [inline src/api/app/services/scm_status_reporter.rb:13]: With the removal of this line, we can also delete the `github?` method since it's not needed anymore.

**Review by dmarcoux — COMMENTED** [review]
I really like the changes in this PR. Good work :+1:

File: src/api/app/services/github_status_reporter.rb
Comment (saraycp) [inline src/api/app/services/github_status_reporter.rb:9]: Before refactoring and around this line we had:
`@workflow_run.save_scm_report_success(request_context) if @workflow_run.present?`

That line has gone, we should bring it back. But let's wait until PR #12799  is merged because it changes that code a bit.

File: src/api/app/services/gitlab_status_reporter.rb
Comment (saraycp) [inline src/api/app/services/gitlab_status_reporter.rb:9]: After this line, we also need to add something like  `@workflow_run.save_scm_report_success(request_context) if @workflow_run.present?`

But let's wait until PR #12799  is merged because it changes that code a bit.

File: src/api/app/services/github_status_reporter.rb
Comment (saraycp) [inline src/api/app/services/github_status_reporter.rb:26]: FYI these lines will be affected by PR #12799

File: src/api/app/services/gitlab_status_reporter.rb
Comment (saraycp) [inline src/api/app/services/gitlab_status_reporter.rb:12]: We need to send metrics to RabbitMq after this rescue. Let's wait until PR #12799  is merged because it changes that code a bit.

**Comment by dmarcoux** [comment]
#12799 has been merged, this PR now has to be rebased.

File: src/api/app/services/github_status_reporter.rb
Comment (saraycp) [inline src/api/app/services/github_status_reporter.rb:27]: This line changed in a previous PR. See [this commit](https://github.com/openSUSE/open-build-service/pull/12786/commits/fb6f6f34ee3230f9e00e7a2c267d9b99d91f3b25)

The specs need to be adapted as well.

**Review by saraycp — APPROVED** [review]

**Pull request merged**
